data/actions/scripts/tools/fishing.lua

```diff
@@ -1,20 +1,93 @@
-local waterIds = {493, 4608, 4609, 4610, 4611, 4612, 4613, 4614, 4615, 4616, 4617, 4618, 4619, 4620, 4621, 4622, 4623, 4624, 4625, 7236, 10499, 15401, 15402}
+-- Water Ids + 1 no fish -1 with fish
+local waterWithFish = {4608, 4610, 4612, 4614, 4616, 4618, 4664, 7236, 15401}
+local waterWithoutFish = {4609, 4611, 4613, 4615, 4617, 4619, 4665, 7237, 15402}
+local waterSpiritId = 10449
+
+local fish = {
+	[1] = {itemid = 2667, baseChance = 25000}, -- Fish : 25%
+	[2] = {itemid = 7159, baseChance = 2250}, -- Green Perch : 2.25%
+	[3] = {itemid = 2669, baseChance = 1100}, -- Northern Pike : 1.1%
+	[4] = {itemid = 2667, baseChance = 700}, -- Rainbow Trout : 0.7%
+	[5] = {itemid = 15405, baseChance = 3000}, -- Sand Fish : 3%
+	[6] = {itemid = 10224, baseChance = 3000} -- Mechanical Fish : 3%
+}
+
 local lootTrash = {2234, 2238, 2376, 2509, 2667}
 local lootCommon = {2152, 2167, 2168, 2669, 7588, 7589}
 local lootRare = {2143, 2146, 2149, 7158, 7159}
 local lootVeryRare = {7632, 7633, 10220}
+
 local useWorms = true
+local useNails = true
+
+local rodId = 2580
+local mechanicalRodId = 10223
 
-local fishers = {}
-local fishingSpotCooldown = 2200 -- Seconds
+local wormId = 3976
+local nailId = 8309
 
 function onUse(player, item, fromPosition, target, toPosition, isHotkey)
-	local targetId = target.itemid
-	if not table.contains(waterIds, targetId) then
-		return false
+	if not target.itemid then return false end
+	
+	-- Normal Fishing --
+	if table.contains(waterWithFish, target.itemid) or table.contains(waterWithoutFish, target.itemid) then
+		if targetId ~= 7236 and targetId ~= 7237 then
```

Comment:
What are those?

Reply:
7236 and 7237 is ice holes. They don't get a magiceffect sent to them when fishing. This code was already in the original.